Core/Goals/GoalFactory.cs

```diff
@@ -57,7 +57,13 @@ public HashSet<GoapGoal> CreateGoals(ClassConfiguration classConfig, IBlacklist
             var combatUtil = new CombatUtil(logger, input, wait, addonReader.PlayerReader);
             var mountHandler = new MountHandler(logger, input, classConfig, wait, addonReader.PlayerReader, stopMoving);
 
-            var targetFinder = new TargetFinder(logger, input, classConfig, wait, addonReader.PlayerReader, blacklist, npcNameTargeting);
+            ITargetFinder targetFinder;
+            if (classConfig.KeyboardOnly)
+            {
+                targetFinder = new TargetFinder(logger, input, classConfig, wait, addonReader.PlayerReader, blacklist, npcNameTargeting);
+            } else {
+                targetFinder = new KeyboardOnlyTargetFinder(logger, input, classConfig, wait, addonReader.PlayerReader, blacklist, npcNameTargeting);
+            }
```

Comment:
I don't think this level of separation is necessary. Please look at the `Core/Goals/TargetFinder/KeyboardOnlyTargetFinder.cs` comment for further explanation.